src/main/java/com/clouway/push/client/ChannelApiPushEventBus.java

```diff
@@ -40,20 +40,15 @@ public HandlerRegistration addHandler(final PushEvent.Type type, final PushEvent
 
     final HandlerRegistration[] handlerRegistration = {null};
 
-    if (!pushChannelApi.hasOpenedChannel()) {
+    if (!pushChannelApi.hasInitialConnection()) {
```

Comment:
Everyone that is using the api, have to make these checks, so I think that we have to move them in the implementation of the PushChannelApi instead of making them outside of the implementation. 

As I can see, we are not handling initial connections differently, so the client code may look like: 

```java
pushChannelApi.connect();
```